plenum/server/consensus/consensus_data_provider.py

```diff
@@ -11,12 +15,17 @@ class ConsensusDataProvider:
     TODO: Consider adding persistent local storage for 3PC certificates
     TODO: Restore primary name from audit ledger instead of passing through constructor
     """
-    def __init__(self, name: str, validators: List[str], primary_name: str):
+
+    def __init__(self, name: str):
```

Comment:
Why remove initial values from constructor?

Reply:
I think that it would be better, to set validators at concrete moments, such as onPoolMembershipChange and getStackParamsAndNodeReg.

Reply:
I agree with your point, however not setting them in constructor actually leaves object half-constructed, which in turn increases chances of bugs. However I'm going to agree if you think it is absolutely necessary to postpone setting validators and current primary (for example, if getting this data at construction time require major refactoring or some ugly hacks).

Reply:
No, i don't see any concerns about initiating it in constructor. Let's make it this way